collector/client.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2016 Mesosphere, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+var (
+	// TODO(roger): place the value of userAgent somewhere convenient
+	userAgent = "com.mesosphere.dcos-metrics/1.0"
+)
+
+// HTTPClient defines the structure of the HTTP client implementation used for
+// collecting metrics from a Mesos master or agent over HTTP.
+type HTTPClient struct {
+	httpClient *http.Client
+	host       string
+	path       string
+	authToken  string // use an empty string "" to disable auth
+}
+
+// NewHTTPClient returns a new instance of the HTTP client.
+func NewHTTPClient(host string, path string, timeout time.Duration) *HTTPClient {
+	var token string
+	// TODO(roger): if auth is required, get a token via jwt/transport. See
+	// https://github.com/dcos/dcos-go/blob/59a7af3/jwt/transport/cmd/example/main.go#L32
+	//
+	// This should be keyed off a configuration option.
+	token = ""
```

Comment:
Since this PR is specifically around the work needed to get the HTTP producer working, I'm inclined to leave this for a separate PR (to prevent this PR from getting any larger than it already is).